File: aulas_praticas/aula2/problema4.py
import re

expr = re.compile(r'(\d{2})(-|\.\.\.|:)(\d{2})\2(\d{2})\2(\d{2})')

with open('input4.txt', 'r', encoding='utf-8') as f:
    content = f.read()
    matriculas = re.finditer(expr, content)
    print([matricula for matricula in matriculas])

File: aulas_praticas/aula2/problema4.js
fs = require("fs");

let content = fs.readFileSync("input4.txt", "utf-8");

content = content.split(/\r?\n/);

const expr = /(\d{2})(-|\.\.\.|:)(\d{2})\2(\d{2})\2(\d{2})/;

for (const line of content) {
    console.log(line.match(expr).map((x) => [...x]));
}
